### PR TITLE
Allow scopes to be applied to system:masters

### LINKED ISSUE #18922: system:masters tokens are not restricted by scopes
While trying to build an integration test for webhooks I came up to the fact that a user that uses a token with system:masters in the groups and scopes has scopes completely ignored and full access a system:masters.

This is in master

### PR DESCRIPTION
**Pull request by enj**
This change overrides the privileged loopback config used by the Kubernetes API server with the config provided by OpenShift's master config.  This causes that config to use certificate based authentication via the system:masters group.  It also prevents the generic API server config completion code from wrapping the OpenShift auth logic with token based loopback support.  While this token was random per master and only persisted in memory, we have no need for it.  This change further adds validation logic that confirms that no completion code warps or alters the OpenShift auth logic.

Signed-off-by: Monis Khan <mkhan@redhat.com>

Fixes #18922
xref: https://github.com/kubernetes/kubernetes/pull/31491

/kind bug
/assign @liggitt @deads2k @simo5
@openshift/sig-security

### CODE REVIEW

File: pkg/cmd/server/origin/master_config.go [hunk 1]
```diff
@@ -219,6 +227,12 @@ func BuildMasterConfig(
 		PrivilegedLoopbackKubernetesClientsetInternal: kubeInternalClient,
 		PrivilegedLoopbackKubernetesClientsetExternal: privilegedLoopbackKubeClientsetExternal,
 
+		// store the original values of the origin auth code for comparison later
+		// this allows use to validate that none of the config completion code
```
Comment (simo5) [inline pkg/cmd/server/origin/master_config.go]: use -> us

**Comment by enj** [comment]
/hold

Want to let @deads2k and @liggitt take a look since some of this is really ugly.

**Comment by deads2k** [comment]
> It also prevents the generic API server config completion code from wrapping the OpenShift auth logic with token based loopback support. While this token was random per master and only persisted in memory, we have no need for it.

/hold

I want to be close to upstream so that our double digits of apiservers are easy to maintain.  I'll look in more detail towards the end of this week.

**Comment by enj** [comment]
Spoke with @deads2k on this.  He plans to kill the cert based loopback for API master soon in favor of the in-memory based token (our API master will be running being the kube API master using a SA so it will not have a cert at all).  We will need to change the upstream code to differentiate between a loopback master call vs `system:masters` (probably using a different custom authorizer) if we want to be able to apply scopes to `system:masters`.

I am going to close this for now, while leaving the issue open so we can revisit at a later date.  Whenever we get this fixed, we can copy the integration test from this PR to validate that our fix works.

**Pull request closed without merging**